src/command_modules/azure-cli-servicebus/azure/cli/command_modules/servicebus/_exception_handler.py

```diff
@@ -0,0 +1,33 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def response_exception_handler(ex):
+    from azure.mgmt.servicebus.models import ErrorResponseException
+    from knack.util import CLIError
+
+    if isinstance(ex, ErrorResponseException):
+        error_message = ex.response.json()
+        error_message = {k.lower(): v for k, v in error_message.items()}
+        if 'error' in error_message:
+            error_message = error_message['error']
+        if 'code' in error_message and 'message' in error_message:
+            message = '{}.'.format(error_message['message']) if error_message['message'] else 'Operation failed.'
+            code = '[Code: "{}"]'.format(error_message['code']) if error_message['code'] else ''
+            raise CLIError('{} - {}'.format(code, message))
+        else:
+            raise CLIError(ex)
+    else:
+        import sys
+        from six import reraise
+        reraise(*sys.exc_info())
+
+
+# pylint: disable=inconsistent-return-statements
+def empty_on_404(ex):
```

Comment:
So for list commands we should show the error instead of showing an empty list.

For example:

```
$az vm list -g foo
Resource group 'foo' could not be found.
```
Please do not use empty_on_404 for your list commands, so that the CLI can have a more consistent behavior for its list commands.

I would also like to add that because of your exception handler, the CLI will give a more detailed error message from the response body; this is still good.

Reply:
removed the list command exceptions handler - empty_on_404

Reply:
Do you also want to remove this function definition?